enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/interceptor/MediationPolicyFilter.java

```diff
@@ -32,51 +39,66 @@
  */
 public class MediationPolicyFilter implements Filter {
     private static final Logger log = LogManager.getLogger(MediationPolicyFilter.class);
+
+    public MediationPolicyFilter() {
+        OPAClient.init();
+    }
+
     @Override
     public boolean handleRequest(RequestContext requestContext) {
         // get operation policies
         PolicyConfig policyConfig = requestContext.getMatchedResourcePath().getPolicyConfig();
         // apply in policies
         if (policyConfig.getIn() != null && policyConfig.getIn().size() > 0) {
             for (Policy policy : policyConfig.getIn()) {
-                applyPolicy(requestContext, policy);
+                if (!applyPolicy(requestContext, policy)) {
+                    return false;
+                }
             }
         }
         return true;
     }
 
-    private void applyPolicy(RequestContext requestContext, Policy policy) {
+    private boolean applyPolicy(RequestContext requestContext, Policy policy) {
         //todo(amali) check policy order
         switch (policy.getAction()) {
             case "SET_HEADER": {
                 addOrModifyHeader(requestContext, policy.getParameters());
-                break;
+                return true;
             }
             case "RENAME_HEADER": {
                 renameHeader(requestContext, policy.getParameters());
-                break;
+                return true;
             }
             case "REMOVE_HEADER": {
                 removeHeader(requestContext, policy.getParameters());
-                break;
+                return true;
             }
             case "ADD_QUERY": {
                 addOrModifyQuery(requestContext, policy.getParameters());
-                break;
+                return true;
             }
             case "REMOVE_QUERY": {
                 removeQuery(requestContext, policy.getParameters());
-                break;
+                return true;
             }
             case "REWRITE_RESOURCE_PATH": {
                 removeAllQueries(requestContext, policy.getParameters());
-                break;
+                return true;
             }
             case "REWRITE_RESOURCE_METHOD": {
                 modifyMethod(requestContext, policy.getParameters());
-                break;
+                return true;
+            }
+            case "OPA": {
+                return opaAuthValidation(requestContext, policy.getParameters());
             }
         }
+
+        // TODO: check placeholders in logs are working
+        log.error("Operation policy action \"{}\" is not supported", policy.getAction(),
+                ErrorDetails.errorLog(LoggingConstants.Severity.MINOR, 6100));
+        return false;
```

Comment:
return 500 to user

Reply:
If the provided action is not matched with any of the policies why should we break the request? if any of the policies is not supported it should be validated in the API deploying time not here imho.

Reply:
+1 The adapter validates policies against policy spec and definition when deploying APIs, but there could be cases where spec and definition may contain invalid actions. We can introduce these action names to Adapter too, so the Adapter also knows what are the supported actions.
We can log as a major (This error is due to our code issue - In adapter, we have not validated the action) error and continue the request flow. @praminda wdyt?

Reply:
The adapter validation will be fixed with https://github.com/wso2/product-microgateway/issues/2692